pkg/steps/lease.go

```diff
@@ -20,6 +20,8 @@ import (
 
 const leaseEnv = "LEASED_RESOURCE"
 
+var noLeaseClientErr = errors.New("invalid configuration, step needs a lease but no lease client provided")
```

Comment:
can we try to suggest where would the the lease client be provided? In ci-op config with `X` stanza?

Reply:
This will only affect people who hand-craft a job that uses a template and does not pass the flags we need, right?

Reply:
I had this sitting around in the comment text box yesterday:

The only two possibilities for this error to occur are:
- `prowgen` doesn't generate the job correctly
- someone writes a job manually

The first has never happened, and it would be on us to fix it anyway.

The second happens when a job is missing the `--lease-…` arguments.  It seemed fragile and a layering violation to add that error message here, so yesterday I worked on changing the way this is handled and validated so it can be detected in a better way, faster, and with better diagnostics.  I'll update this PR later today.